Document bypassing CI (#8569)
ppy/osu-wiki: wiki/osu!_wiki/Maintenance/en.md

The osu! wiki repository uses [continuous integration](https://docs.github.com/en/actions/guides/about-continuous-integration) (CI) to automatically check incoming pull requests for various common errors. The list of checks is configured in the [`continuous-integration.yml`](https://github.com/ppy/osu-wiki/blob/master/.github/workflows/continuous-integration.yml) file. 

> The [`package.json`](https://github.com/ppy/osu-wiki/blob/master/package.json) file lists all (remark) plugins used by the CI, of which some were written by the osu! wiki maintainers.

Reviewer: "all (remark) plugins" reads weird for one reason or another, maybe because it's not quite straightforward it's a tool name

Author: would it be better to link it? most of this article is [signposted as being technical](https://github.com/ppy/osu-wiki/blob/89e54bd9cea940104d678ce060343535e23eaee2/wiki/osu!_wiki/Maintenance/en.md?plain=1#L15) anyway

```suggestion
The [`package.json`](https://github.com/ppy/osu-wiki/blob/master/package.json) file lists all ([remark](https://github.com/remarkjs/remark)) plugins used by the CI, of which some were written by the osu! wiki maintainers.
```

Reviewer: yes that gives enough context

The CI checks are run automatically on every commit of a recurring contributor. In order to have their pull requests merged, contributors are expected to fix errors reported by the CI. To see the [status of checks](img/ci-status.png), do the following:

1. Scroll down the pull request's page, find the `osu-wiki continuous integration` status row, and click the `Details` link.
2. On the new page, expand the `run remark on changed files` step. Each finding is accompanied by its exact location in a file and a short description of why it is an error.

Some errors also appear as annotations on the `Files changed` tab below the line where the issue was found.

If you need help with decrypting CI check error messages, or fixing issues, ask in the `#osu-wiki` channel on Discord.

#### Bypassing CI checks

> CI checks ensure that changes to the wiki keep it in a consistent state, but ways to bypass them exist as a fail safe, such that pull requests can still be merged in the event of false errors or discovered bugs in the checks. There are a few situations where intentionally bypassing a CI check is acceptable, as outlined below. Contact a [maintainer](/wiki/osu!_wiki/Maintenance/List_of_maintainers) if you need to bypass a check for a reason not mentioned here.

Reviewer: I feel the first sentence is trying to say too much (=overloaded), mind simplifying or at least chopping it down a bit?

Author: perhaps like this?

```suggestion
CI checks normally prevent pull requests with errors from getting merged. However, in the event of false errors or discovered bugs in the checks, they can still be bypassed as a fail safe. There are a few situations where intentionally bypassing a CI check is acceptable, as outlined below. Contact a [maintainer](/wiki/osu!_wiki/Maintenance/List_of_maintainers) if you need to bypass a check for a reason not mentioned here.
```

Reviewer: yes!

| 4 | Broken wiki links | `check-links` command of [`osu-wiki-tools`](https://github.com/Walavouchey/osu-wiki-tools) | Whether internal [wiki links](/wiki/Article_styling_criteria/Formatting#wiki-links) point to an actual article, news post (for news post links), or section thereof. | Add `SKIP_WIKILINK_CHECK` anywhere in the pull request description. |
| 5 | Outdated translations | `check-outdated-articles` command of [`osu-wiki-tools`](https://github.com/Walavouchey/osu-wiki-tools) | Whether translations are properly [marked as outdated](/wiki/Article_styling_criteria/Formatting#outdated-translations) when updating an English article. | Add `SKIP_OUTDATED_CHECK` anywhere in the pull request description. |

##### Markdown [`no-heading-punctuation`](https://github.com/remarkjs/remark-lint/tree/main/packages/remark-lint-no-heading-punctuation) remark lint rule

> Trailing periods in headings normally do not occur because headings are generally not sentences, and are therefore disallowed. Sometimes trailing punctuation is needed because e.g. a song title in the heading contains it, or a sentence is put into a heading in an FAQ section.

Reviewer: "or a sentence is put into a heading" is I think the reason this check exists, so no. if it "is put into a heading", the trailing period still has to go
```suggestion
Trailing periods in headings normally do not occur because headings are generally not sentences, and are therefore disallowed. Sometimes trailing punctuation is needed because e.g. a song title in the heading contains it.
```

Author: for full sentences, that's basically the only place you'd use it, along with trailing questions marks. i'm kind of referring to [this currently sole example](https://github.com/ppy/osu-wiki/blob/1478755324ad093078d29949579b286f7b232384/wiki/Help_centre/Upgrading_to_lazer/en.md?plain=1#L150-L152) (in all other faq sections, the headings end in `?` or `!`)

```suggestion
Trailing periods in headings normally do not occur because headings are generally not sentences, and are therefore disallowed. Sometimes trailing punctuation may be needed because e.g. a song title or artist name contains it, or for an FAQ-style section.
```

Author: hmm i'll leave it like this then

```suggestion
Trailing periods in headings normally do not occur because headings are generally not sentences, and are therefore disallowed. However, sometimes trailing punctuation may be needed because e.g. a song title or artist name contains it.
```

```markdown
<!-- lint ignore no-heading-punctuation -->

### Amusing Reflection Rag.
```

##### Markdown [`heading-increment`](https://github.com/remarkjs/remark-lint/tree/main/packages/remark-lint-heading-increment) remark lint rule

Normally, heading levels must only ever increase by one at a time. For [infoboxes](/wiki/Article_styling_criteria/Formatting#infoboxes) however, only heading levels of 4 and 5 are allowed, which may conflict with this rule.

```markdown
# List of peppy's favourite mappers

::: Infobox
<!-- lint ignore heading-increment -->

#### peppy

Creator of osu!.
:::
```

##### Wiki link check

*See also: [Article styling criteria/Formatting § Wiki links](/wiki/Article_styling_criteria/Formatting#wiki-links)*

Contributors are encouraged to fix broken links while updating articles. That said, for some situations, that may require more work than is relevant for the pull request:

> - Small individual fixes to an article (that are not meant to generally clean up an article)

Author: ```suggestion
- Small individual fixes to articles (that are not meant to generally clean them up)
```
